core/src/main/scala/org/apache/spark/sql/delta/Snapshot.scala

```diff
@@ -286,16 +286,22 @@ class Snapshot(
   lazy val numIndexedCols: Int = DeltaConfigs.DATA_SKIPPING_NUM_INDEXED_COLS.fromMetaData(metadata)
 
   /** Return the set of properties of the table. */
-  def getProperties: mutable.HashMap[String, String] = {
-    val base = new mutable.HashMap[String, String]()
+  def getProperties: mutable.Map[String, String] = {
```

Comment:
why this change?

Reply:
Because we want to keep enabled features at the end of the prop list, otherwise too many enabled features will make the output unreadable. To do this we have to use `LinkedHashMap` which does keep the insertion order: https://www.scala-lang.org/api/2.13.6/scala/collection/mutable/LinkedHashMap.html.
Meanwhile we do not want to break the ABI compatibility, thus keeping the return `mutable.HashMap` interface unchanged.